library/tabling.pl

```diff
@@ -179,7 +179,8 @@
     update/4.
 
 update(M:Wrapper, A1, A2, A3) :-
-    M:'$table_update'(Wrapper, A1, A2, A3).
+    M:'$table_update'(Wrapper, A1, A2, A3),
+    A1\==A3.
```

Comment:
Shouldn't this be `A1 =@= A2`, as tabling is only about variants?

Reply:
Sorry, `\+ A1 =@= A2`?

Reply:
Yes, better